metric-anomaly-detector/src/test/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricQueryBuilderTest.java

```diff
@@ -77,4 +78,72 @@ void testBuildMetricQueryRequest() {
 
     Assertions.assertNotNull(queryRequest);
   }
+
+  @Test
+  void testAttributeCaching() {
+    LhsExpression lhsExpression =
+        LhsExpression.newBuilder()
+            .setAttribute(Attribute.newBuilder().setKey("id").setScope("SERVICE").build())
+            .build();
+    RhsExpression rhsExpression = RhsExpression.newBuilder().setStringValue("1234").build();
+    LeafFilter leafFilter =
+        LeafFilter.newBuilder()
+            .setValueOperator(ValueOperator.VALUE_OPERATOR_EQ)
+            .setLhsExpression(lhsExpression)
+            .setRhsExpression(rhsExpression)
+            .build();
+
+    MetricSelection metricSelection =
+        MetricSelection.newBuilder()
+            .setMetricAggregationInterval("PT15s")
+            .setMetricAggregationFunction(
+                MetricAggregationFunction.METRIC_AGGREGATION_FUNCTION_TYPE_SUM)
+            .setFilter(Filter.newBuilder().setLeafFilter(leafFilter).build())
+            .setMetricAttribute(
+                Attribute.newBuilder().setKey("duration").setScope("SERVICE").build())
+            .build();
+
+    AttributeServiceClient attributesServiceClient = mock(AttributeServiceClient.class);
+    List<AttributeMetadata> attributesList1 =
+        List.of(
+            AttributeMetadata.newBuilder()
+                .setScopeString(AttributeScope.SERVICE.name())
+                .setKey("id")
+                .setId("Service.id")
+                .build(),
+            AttributeMetadata.newBuilder()
+                .setScopeString(AttributeScope.SERVICE.name())
+                .setKey("startTime")
+                .setId("Service.startTime")
+                .build());
+    when(attributesServiceClient.findAttributes(
+            eq(Map.of("x-tenant-id", "__default")),
+            eq(
+                AttributeMetadataFilter.newBuilder()
+                    .addScopeString(AttributeScope.SERVICE.name())
+                    .build())))
+        .thenAnswer((Answer<Iterator<AttributeMetadata>>) invocation -> attributesList1.iterator());
+
+    MetricQueryBuilder mqb = new MetricQueryBuilder(attributesServiceClient);
+    CacheStats stats = mqb.attributeServiceCache.stats();
+
+    // Cache stats before adding any key
+    Assertions.assertEquals(0, stats.missCount());
+    Assertions.assertEquals(0, stats.hitCount());
+
+    String beforeVal = mqb.getTimestampAttributeId("__default", "SERVICE");
```

Comment:
Can we directly call the `cache` here instead of calling `getTimestampAttributeId` ?

Reply:
we want to test the method which makes the cache call